### PR TITLE
DDF-1465 - added case sensitive search in the search ui via the Match…

### PR DESCRIPTION
**Pull request by kcwire**
… Case checkbox

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/272)

### CODE REVIEW

**Comment by kcwire** [comment]
@jrnorth is hero
@beyelerb 
@EIrwin 
@jlcsmith

**Comment by jrnorth** [comment]
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

<sup>**[catalog/ui/search-ui/standard/src/main/webapp/js/model/Query.js, line 318 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/272#-K15rhrMGZXxoirm-jSP)** ([raw file](https://github.com/codice/ddf/blob/7d8b1d34a7062283ff6849bbad10f3d9ec6b9264/catalog/ui/search-ui/standard/src/main/webapp/js/model/Query.js#L318)):</sup>
Could simplify this to `var likeOp = this.get('matchcase') ? ' LIKE ' : ' ILIKE ';`


---

<sup>**[catalog/ui/search-ui/standard/src/main/webapp/templates/search/searchForm.handlebars, line 41 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/272#-K15tiy6FtjUHZCUK35t)** ([raw file](https://github.com/codice/ddf/blob/7d8b1d34a7062283ff6849bbad10f3d9ec6b9264/catalog/ui/search-ui/standard/src/main/webapp/templates/search/searchForm.handlebars#L41)):</sup>
Would "Case Sensitive" be more intuitive to the user than "Match Case"?


---

<sup>**[catalog/ui/search-ui/standard/src/main/webapp/templates/search/searchForm.handlebars, line 47 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/272#-K15sWLzxrKEBiKgZQ-K)** ([raw file](https://github.com/codice/ddf/blob/7d8b1d34a7062283ff6849bbad10f3d9ec6b9264/catalog/ui/search-ui/standard/src/main/webapp/templates/search/searchForm.handlebars#L47)):</sup>
Might as well get rid of the `<div>` below it since it doesn't serve a purpose now. It looks like we're using this `clear:both` pattern in a few other places in this file, so it may be worth updating them to use the same strategy as the one being used here (i.e., putting the `clear:both` on the `<li>` rather than a separate `<div>`).  At the least, it would make things more consistent (and logically, it seems to make more sense since we really don't need these extra `<div>`s as far as I can tell).


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/272)

**Comment by kcwire** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

<sup>**[catalog/ui/search-ui/standard/src/main/webapp/js/model/Query.js, line 318 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/272#-K15rhrMGZXxoirm-jSP)** ([raw file](https://github.com/codice/ddf/blob/7d8b1d34a7062283ff6849bbad10f3d9ec6b9264/catalog/ui/search-ui/standard/src/main/webapp/js/model/Query.js#L318)):</sup>
done as suggested.


---

<sup>**[catalog/ui/search-ui/standard/src/main/webapp/templates/search/searchForm.handlebars, line 41 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/272#-K15tiy6FtjUHZCUK35t)** ([raw file](https://github.com/codice/ddf/blob/7d8b1d34a7062283ff6849bbad10f3d9ec6b9264/catalog/ui/search-ui/standard/src/main/webapp/templates/search/searchForm.handlebars#L41)):</sup>
I thought "Match Case" might make more sense to non developers.  I could be persuaded if someone has a string opinion.


---

<sup>**[catalog/ui/search-ui/standard/src/main/webapp/templates/search/searchForm.handlebars, line 47 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/272#-K15sWLzxrKEBiKgZQ-K)** ([raw file](https://github.com/codice/ddf/blob/7d8b1d34a7062283ff6849bbad10f3d9ec6b9264/catalog/ui/search-ui/standard/src/main/webapp/templates/search/searchForm.handlebars#L47)):</sup>
done. Got rid of divs that just contained the `style="clear:both"`.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/272)

File: catalog/ui/search-ui/standard/src/main/webapp/js/model/Query.js [hunk 1]
```diff
@@ -175,12 +175,21 @@ define([
                 // contextual
                 var q = this.get('q');
                 if (q) {
-                    filters.push(new Filter.Model({
-                        fieldName: 'anyText',
-                        fieldType: 'string',
-                        fieldOperator: 'contains',
-                        stringValue1: q
-                    }));
+                    if (this.get('matchcase')){
```
Comment (coyotesqrl) [inline catalog/ui/search-ui/standard/src/main/webapp/js/model/Query.js]: Maybe just a ternary?
```
if (q) {
  filters.push(new Filter.Model({
    fieldName: 'anyText',
    fieldType: 'string',
    fieldOperator: (this.get('matchcase') ? 'matchcase' : 'contains',
    stringValue1: q
  }));
```

**Comment by jrnorth** [comment]
Hero build successful.

I uploaded two Word documents via the Search UI: one with the text "The Quick Brown Fox Jumped Over The Lazy Dog," and one with text "the quick brown fox jumped over the lazy dog". The metacard titles, respectively, were changed to "Word Document" and "word document".

Manual tests:
* On the main search page, the queries "Quick Brown Fox", "quick brown fox", and "Quick Brown fox" all returned the correct results with the "Match Case" checkbox checked and unchecked
* On the enhanced search page, verified that the "metadata" field (an xml field) and the "title" field (a string field) both had "matchcase" as an operator option.
* On the enhanced search page, the queries "Quick Brown Fox", "quick brown fox", and "Quick Brown fox" on the "metadata" field returned the correct results with the "contains" operator selected and with the "matchcase" operator selected
* On the enhanced search page, the queries "Word", "word", and "Word document" on the "title" field returned the correct results with the "contains" operator selected and with the "matchcase" operator selected
* Verified on Chrome, Firefox, and IE

**Pull request merged**